### PR TITLE
MWPW-144756 Table - Responsive breakpoints - Text wraps under icon in Content cell

### PR DESCRIPTION
**Pull request by sivasadobe**
* Changed the Icon and Text element as a flex box.
* The content cell's text element will be wrapped inside the corresponding box
* Icons & Text are vertically centered

Resolves: [MWPW-144756](https://jira.corp.adobe.com/browse/MWPW-144756)

## Test URLs:
**CC**
- Before: https://main--milo--adobecom.hlx.live/drafts/siva/plans?martech=off
- After: https://mwpw-144756--milo--sivasadobe.hlx.live/drafts/siva/plans?martech=off

**Milo Table Doc**

- Before: https://main--milo--adobecom.hlx.live/docs/authoring/tables
- After: https://mwpw-144756--milo--sivasadobe.hlx.live/docs/authoring/tables

### CODE REVIEW

**Comment by spadmasa** [comment]
@sivasadobe  can you please check the alignment is not seen fine , in the feature branch appended

https://mwpw-144756--milo--sivasadobe.hlx.page/drafts/siva/plans?martech=off
https://main--cc--adobecom.hlx.page/products/firefly/plans?milolibs=mwpw-144756--milo--sivasadobe
image

CC : @sigadamvenkata

**Comment by sivasadobe** [comment]
> @sivasadobe can you please check the alignment is not seen fine , in the feature branch appended
> 
> https://mwpw-144756--milo--sivasadobe.hlx.page/drafts/siva/plans?martech=off https://main--cc--adobecom.hlx.page/products/firefly/plans?milolibs=mwpw-144756--milo--sivasadobe image
> 
> CC : @sigadamvenkata

@spadmasa The texts are properly aligned. Could you check again

**Comment by spadmasa** [comment]
@sivasadobe i still see the same issue , please check 
image

**Comment by sivasadobe** [comment]
The second line of text is properly aligned beneath the text.
https://main--cc--adobecom.hlx.page/products/firefly/plans?milolibs=mwpw-144756--milo--sivasadobe

main--cc--adobecom hlx page_products_firefly_plans_milolibs=mwpw-144756--milo--sivasadobe

cc @spadmasa

**Comment by Anupamasonnalagi** [comment]
Verified for the below URL's.
https://mwpw-144756--milo--sivasadobe.hlx.page/drafts/siva/plans?martech=off&georouting=off
image
https://main--cc--adobecom.hlx.page/products/firefly/plans?milolibs=mwpw-144756--milo--sivasadobe
IMG_0191
IMG_3780

**Comment by sivasadobe** [comment]
Pushed code changes to support various icons, not just the :checkmark:

cc @spadmasa @Anupamasonnalagi

**Comment by Anupamasonnalagi** [comment]
Verified the fix for various icons. It's fine.
https://mwpw-144756--milo--sivasadobe.hlx.page/drafts/Anupama/pr/plans
IMG_3785
image

File: libs/blocks/table/table.js [hunk 1]
```diff
@@ -140,10 +140,17 @@ function handleTitleText(cell) {
   if (cell.querySelector('.table-title-text')) return;
   const textSpan = createTag('span', { class: 'table-title-text' });
```
Comment (mokimo) [inline libs/blocks/table/table.js:141]: I would suggest something such as
```js
const textSpan = createTag('span', { class: 'table-title-text' });
while (cell.firstChild) textSpan.append(cell.firstChild);

const iconTooltip = textSpan.querySelector('.icon-info, .icon-tooltip, .milo-tooltip');
if (iconTooltip) cell.append(iconTooltip.closest('em') || iconTooltip);

const firstIcon = textSpan.querySelector('.icon:first-child');
let nodeToInsert = textSpan;

if (firstIcon) {
  const titleRowSpan = createTag('span', { class: 'table-title-row' });
  titleRowSpan.append(firstIcon, textSpan);
  nodeToInsert = titleRowSpan;
}

cell.insertBefore(nodeToInsert, cell.firstChild);
```

**Review by mokimo — APPROVED** [review]
You can simplify the if/else statement, other than that nice 👍

File: libs/blocks/table/table.js (hunk 1 above)
Comment (sivasadobe) [reply libs/blocks/table/table.js:141]: @mokimo , I changed the code.
Thank you for the suggestion; it now looks good.

**Comment by spadmasa** [comment]
Validation done on the below url 

https://main--cc--adobecom.hlx.page/products/firefly/plans?milolibs=mwpw-144756--milo--sivasadobe

image
image
image

Verified for other icons as well 
https://main--cc--adobecom.hlx.page/drafts/Souj/plans?milolibs=mwpw-144756--milo--sivasadobe&martech=off
https://mwpw-144756--milo--sivasadobe.hlx.page/drafts/siva/plans?martech=off

**Comment by mokimo** [comment]
@sivasadobe it seems that the PSI check is failing - the automation process won't pick this PR up if the checks are not passing.
see https://github.com/adobecom/milo/pull/2230#issuecomment-2087806049

**Pull request merged**